pieforproviders/pieforproviders
# Feature/54/api call

# 💅🏼 What issue does this fix?
This is the last piece of #54 - users can create an account and an email will be sent to their email (or displayed in the log in dev environments)

This PR:
* reorders attributes alphabetically
* renames the columns for phone from phone/mobile to phone_number/phone_type
* adds a confirmation page and redirects to it after successful user creation
* adds a wrapper hook for API calls so we don't have to redefine error messages everywhere
* transforms data keys in the JS request to snake case for compatibility with rails using an initializer
* adds a custom MIME type for our Accept header (this might not have been necessary, but I don't want to rabbit hole further on it)
* styles inactive buttons
* makes phone numbers unique

## 🍂 Before You Submit


## 🛷 Deployment Considerations


## 🧵 Steps to set up locally

Restart your rails server

## 🧳 Steps to test

* visit localhost:3000/signup
* fill out all the fields and click "submit"; you should be redirected to the confirmation page and your user should be created

## 🕯 Caveats, concerns
This does not take care of error handling on failed user creation for reasons of duplicate users
